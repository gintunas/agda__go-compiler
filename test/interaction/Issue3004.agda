
module _ where

module Test₁ where
  postulate
    id : {X : Set} → X → X
    A : Set
    x : A

  record S : Set where
    field
      a : A

  postulate
    B : S → Set

  record T : Set where
    field
      s : S
      b : B s

  -- Agda hangs here
  t : T
  t = λ
    { .T.s .S.a → x
    ; .T.b      → id {!!}
    }

module Test₂ where
  postulate
    id : (X : Set) → X → X
    A : Set
    B : A → Set
    x : A

  record T : Set where
    field
      a : A
      b : B a

  -- Agda hangs here
  t : T
  t = λ
    { .T.a → x
    ; .T.b → id {!!} {!!}
    }
