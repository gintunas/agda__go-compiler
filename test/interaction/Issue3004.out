(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : B (T.s (λ { .T.s .S.a → x ; .T.b → id ?0 })) ?1 : Set ?2 : B (T.a (λ { .T.a → x ; .T.b → id (B (T.a .extendedlambda0)) ?2 })) " nil)
((last . 1) . (agda2-goals-action '(0 1 2)))
